Engine/DataFeeds/Enumerators/Factories/TimeTriggeredUniverseSubscriptionEnumeratorFactory.cs

```diff
@@ -52,8 +59,82 @@ public TimeTriggeredUniverseSubscriptionEnumeratorFactory(ITimeTriggeredUniverse
         /// <returns>An enumerator reading the subscription request</returns>
         public IEnumerator<BaseData> CreateEnumerator(SubscriptionRequest request, IDataProvider dataProvider)
         {
-            return _universe.GetTriggerTimes(request.StartTimeUtc, request.EndTimeUtc, _marketHoursDatabase)
-                .Select(x => new Tick {Time = x, Symbol = request.Configuration.Symbol}).GetEnumerator();
+            var enumerator = (IEnumerator<BaseData>) _universe.GetTriggerTimes(request.StartTimeUtc, request.EndTimeUtc, _marketHoursDatabase)
+                .Select(x => new Tick { Time = x, Symbol = request.Configuration.Symbol })
+                .GetEnumerator();
+
+            var universe = request.Universe as UserDefinedUniverse;
+            if (universe != null)
+            {
+                enumerator = new InjectionEnumerator(enumerator);
+
+                // Trigger universe selection when security added/removed after Initialize
+                universe.CollectionChanged += (sender, args) =>
+                {
+                    var items =
+                        args.Action == NotifyCollectionChangedAction.Add ? args.NewItems :
+                        args.Action == NotifyCollectionChangedAction.Remove ? args.OldItems : null;
+
+                    var time = _timeProvider.GetUtcNow();
+                    if (items == null || time == DateTime.MinValue) return;
+
+                    var symbol = items.OfType<Symbol>().FirstOrDefault();
+
+                    if(symbol == null) return;
+
+                    // the data point time should always be in exchange timezone
+                    time = time.ConvertFromUtc(request.Configuration.ExchangeTimeZone);
+
+                    var collection = new BaseDataCollection(time, symbol);
+
+                    ((InjectionEnumerator) enumerator).InjectDataPoint(collection);
+                };
+            }
+
+            return enumerator;
+        }
+
+        private class InjectionEnumerator : IEnumerator<BaseData>
```

Comment:
I've grown more comfortable with this guy's implementation over the last couple days. Nice work 👍